lib/sandbox/pmapi.js

```diff
@@ -44,9 +44,16 @@ Postman = function Postman (bridge, execution, onRequest) {
         iterationData;
 
     // @todo - ensure runtime passes data in a scope format
-    iterationData = new VariableScope();
+    iterationData = new VariableScope(execution.data);
     iterationData.syncVariablesFrom(execution.data);
 
+    // instead of creating new instance of variableScope,
+    // reuse one so that any changes made through pm.variables.set() is directly reflected
+    execution.variables.addLayer(iterationData.values);
```

Comment:
@shamasis we are reconstructing layers on sandbox because `variableScope.toJSON` removes layers. All variables passed to sandbox context are `toJSON`ed while sending through bridge events.

Reply:
this is a better approach.